JSDoc `@import` of "tslib" fails under `--moduleResolution bundler`

```ts
// @noEmit: true
// @checkJs: true
// @allowJs: true
// @module: preserve

// @Filename: node_modules/tslib/package.json
{
    "name": "tslib",
    "exports": {
        ".": {
            "module": {
                "types": "./modules/index.d.ts",
                "default": "./tslib.es6.mjs"
            },
            "import": {
                "node": "./modules/index.js",
                "default": {
                    "types": "./modules/index.d.ts",
                    "default": "./tslib.es6.mjs"
                }
            },
            "default": "./tslib.js"
        },
        "./*": "./*",
        "./": "./"
    }
}

// @Filename: node_modules/tslib/modules/index.d.ts
export {};

// @Filename: node_modules/tslib/tslib.d.ts
export {};

// @Filename: test.js
/** @import * as tslib from "tslib" */
/** @type {typeof tslib} T */
```

## Expected behavior

Import resolves

## Actual behavior

```
Cannot find module 'tslib' or its corresponding type declarations.ts(2307)
```

Even though quick info also shows

```ts
module "/Users/andrew/Developer/microsoft/eg/js/node_modules/tslib/modules/index"
```

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -4617,12 +4617,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             ? location
             : (isModuleDeclaration(location) ? location : location.parent && isModuleDeclaration(location.parent) && location.parent.name === location ? location.parent : undefined)?.name ||
                 (isLiteralImportTypeNode(location) ? location : undefined)?.argument.literal ||
-                (isInJSFile(location) && isJSDocImportTag(location) ? location.moduleSpecifier : undefined) ||
                 (isVariableDeclaration(location) && location.initializer && isRequireCall(location.initializer, /*requireStringLiteralLikeArgument*/ true) ? location.initializer.arguments[0] : undefined) ||
                 findAncestor(location, isImportCall)?.arguments[0] ||
-                findAncestor(location, isImportDeclaration)?.moduleSpecifier ||
-                findAncestor(location, isExternalModuleImportEqualsDeclaration)?.moduleReference.expression ||
-                findAncestor(location, isExportDeclaration)?.moduleSpecifier;
+                findAncestor(location, or(isImportDeclaration, isJSDocImportTag, isExportDeclaration))?.moduleSpecifier ||
+                findAncestor(location, isExternalModuleImportEqualsDeclaration)?.moduleReference.expression;
         const mode = contextSpecifier && isStringLiteralLike(contextSpecifier)
             ? host.getModeForUsageLocation(currentSourceFile, contextSpecifier)
             : host.getDefaultResolutionModeForFile(currentSourceFile);

--- a/tests/baselines/reference/importTag22.errors.txt
+++ b/tests/baselines/reference/importTag22.errors.txt
@@ -1,0 +1,38 @@
+test.ts(2,17): error TS2304: Cannot find name 'tslib'.
+
+
+==== node_modules/tslib/package.json (0 errors) ====
+    {
+      "name": "tslib",
+      "exports": {
+          ".": {
+              "module": {
+                  "types": "./modules/index.d.ts",
+                  "default": "./tslib.es6.mjs"
+              },
+              "import": {
+                  "node": "./modules/index.js",
+                  "default": {
+                      "types": "./modules/index.d.ts",
+                      "default": "./tslib.es6.mjs"
+                  }
+              },
+              "default": "./tslib.js"
+          },
+          "./*": "./*",
+          "./": "./"
+      }
+    }
+    
+==== node_modules/tslib/modules/index.d.ts (0 errors) ====
+    export {};
+    
+==== node_modules/tslib/tslib.d.ts (0 errors) ====
+    export {};
+    
+==== test.ts (1 errors) ====
+    /** @import * as tslib from "tslib" */
+    type T = typeof tslib
+                    ~~~~~
+!!! error TS2304: Cannot find name 'tslib'.
+    

--- a/tests/cases/conformance/jsdoc/importTag21.ts
+++ b/tests/cases/conformance/jsdoc/importTag21.ts
@@ -1,0 +1,38 @@
+// @noEmit: true
+// @checkJs: true
+// @allowJs: true
+// @module: preserve
+// @noTypesAndSymbols: true
+
+// @Filename: node_modules/tslib/package.json
+{
+  "name": "tslib",
+  "exports": {
+      ".": {
+          "module": {
+              "types": "./modules/index.d.ts",
+              "default": "./tslib.es6.mjs"
+          },
+          "import": {
+              "node": "./modules/index.js",
+              "default": {
+                  "types": "./modules/index.d.ts",
+                  "default": "./tslib.es6.mjs"
+              }
+          },
+          "default": "./tslib.js"
+      },
+      "./*": "./*",
+      "./": "./"
+  }
+}
+
+// @Filename: node_modules/tslib/modules/index.d.ts
+export {};
+
+// @Filename: node_modules/tslib/tslib.d.ts
+export {};
+
+// @Filename: test.js
+/** @import * as tslib from "tslib" */
+/** @type {typeof tslib} T */

--- a/tests/cases/conformance/jsdoc/importTag22.ts
+++ b/tests/cases/conformance/jsdoc/importTag22.ts
@@ -1,0 +1,36 @@
+// @noEmit: true
+// @module: preserve
+// @noTypesAndSymbols: true
+
+// @Filename: node_modules/tslib/package.json
+{
+  "name": "tslib",
+  "exports": {
+      ".": {
+          "module": {
+              "types": "./modules/index.d.ts",
+              "default": "./tslib.es6.mjs"
+          },
+          "import": {
+              "node": "./modules/index.js",
+              "default": {
+                  "types": "./modules/index.d.ts",
+                  "default": "./tslib.es6.mjs"
+              }
+          },
+          "default": "./tslib.js"
+      },
+      "./*": "./*",
+      "./": "./"
+  }
+}
+
+// @Filename: node_modules/tslib/modules/index.d.ts
+export {};
+
+// @Filename: node_modules/tslib/tslib.d.ts
+export {};
+
+// @Filename: test.ts
+/** @import * as tslib from "tslib" */
+type T = typeof tslib

